library/std/src/f64/tests.rs

```diff
@@ -635,6 +635,37 @@ fn test_atanh() {
     assert_approx_eq!((-0.5f64).atanh(), -0.54930614433405484569762261846126285f64);
 }
 
+#[test]
+fn test_gamma() {
+    assert_approx_eq!(1.0f64.gamma(), 1.0f64);
+    assert_approx_eq!(2.0f64.gamma(), 1.0f64);
+    assert_approx_eq!(3.0f64.gamma(), 2.0f64);
+    assert_approx_eq!(4.0f64.gamma(), 6.0f64);
+    assert_approx_eq!(5.0f64.gamma(), 24.0f64);
```

Comment:
...for what platform is this ever imprecise?

Reply:
Not sure what guarantees different platforms provide around preciseness (or if it can change over time), but we can try with `assert_eq!`

Reply:
Looks like it fails on `x86_64-unknown-linux-gnu`

Reply:
My disappointment is immeasurable and my day is ruined.

Well, okay. Can you add a comment to this effect and then squash down the first 6 commits before the one that includes the `compiler_builtins` bump?

Reply:
Updated and squashed

Reply:
No problem, thanks for helping move this along!